File: CW3Java/src/main/java/io/rienel/task1/OfferService.java
package io.rienel.task1;

import io.rienel.task1.model.Client;
import io.rienel.task1.model.Offer;
import io.rienel.task1.model.Stuff;

import java.time.LocalDate;
import java.util.List;

public interface OfferService {
    Offer signNewOffer(LocalDate startDate, LocalDate endingDate,
                       Client client, Stuff stuff);

    Offer findOfferBySerialNumber(String serialNumber);

    List<Offer> findOffersByClientSurname(String clientSurname);

    List<Offer> findOffersByStuff(Stuff stuff);
}

File: CW3Kotlin/src/main/kotlin/io/rienel/task1/OfferService.kt
package io.rienel.task1

import io.rienel.task1.model.Client
import io.rienel.task1.model.Offer
import io.rienel.task1.model.Stuff
import java.time.LocalDate

interface OfferService {
    fun signNewOffer(startDate: LocalDate, endingDate: LocalDate, client: Client, stuff: Stuff): Offer

    fun findOfferBySerialNumber(serialNumber: String): Offer

    fun findOffersByClientSurname(surname: String): List<Offer>

    fun findOffersByStuff(stuff: Stuff): List<Offer>
}